addons/account/models/chart_template.py

```diff
@@ -578,6 +582,131 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
+            # If nothing is found, don't fill it to avoid setting a wrong account
+            similar_repartition_line = None
+            while not similar_repartition_line and additional_domain:
+                search_domain = minimal_domain + additional_domain
+                similar_repartition_line = self.env['account.tax.repartition.line'].search(search_domain, limit=1)
+                additional_domain.pop()
+
+            similar_account_id = None
+            if similar_repartition_line:
+                local_tax_account = similar_repartition_line.account_id
+                similar_account_id = create_foreign_tax_account(local_tax_account, _("Foreign tax account (%s)", country.code)).id
+
+            existing_accounts[template_id] = similar_account_id
+            return existing_accounts[template_id]
+
+        # Retrieve the required l10n datas
+        chart_template_code = self._guess_chart_template(country=country)
+        tax_groups = self._get_chart_template_data(chart_template_code)['account.tax.group']
+        taxes = self._get_chart_template_data(chart_template_code)['account.tax']
+
+        # try to replace tax group default accounts by a proper one or clean the given xml_id as it doesn't exist
+        # in the company's chart template
+        property_name = (
+            ('tax_payable_account_id', _("Foreign tax account payable (%s)", country.code)),
+            ('tax_receivable_account_id', _("Foreign tax account receivable (%s)", country.code)),
+            ('advance_tax_payment_account_id', _("Foreign tax account advance payment (%s)", country.code)),
+        )
+        property_company = self.env['ir.property'].with_company(company)
+        for account_field, account_name in property_name:
+            property_account_field = f"property_{account_field}"
+            default_acc = property_company._get(property_account_field, 'account.tax.group')
+            foreign_account_id = create_foreign_tax_account(default_acc, account_name).id if default_acc else False
+            for tax_group in tax_groups.values():
+                tax_group[account_field] = foreign_account_id
+
+        for tax_template in taxes.values():
+            tax_template['country_id'] = country.id  # This is required because the country isn't provided directly by the template
+
+            # Create and set accounts corresponding to the current chart template
+            for _command, _id, rep_line in tax_template.get('repartition_line_ids', []):
+                if rep_line['repartition_type'] == 'tax' and 'account_id' in rep_line:
+                    rep_line['account_id'] = corresponding_tax_account(tax_template['type_tax_use'], rep_line)
+
+            if 'cash_basis_transition_account_id' in tax_template:
+                transition_account_template = tax_template['cash_basis_transition_account_id']
+                corresponding_transition_account = existing_accounts.get(transition_account_template, False)
```

Comment:
Shouldn't everything related to the cache of `corresponding_tax_account` be managed inside of `corresponding_tax_account`?

Reply:
I think I didn't do it because the function you talked about is for finding account related to repartition.line while this one is related to the taxes.
I will recheck tomorrow.

Tips for my futur self: maybe it would be better that the function provides all the accounts at once for one tax.